app/lib/server/functions/getUserSingleOwnedRooms.js

```diff
@@ -0,0 +1,78 @@
+import { hasRole, getUsersInRole } from '../../../authorization';
+import { Users, Subscriptions, Rooms } from '../../../models';
+
+export const getUserSingleOwnedRooms = function(userId) {
+	const roomsThatWillChangeOwner = [];
+	const roomsThatWillBeRemoved = [];
+
+	// Iterate through all the rooms the user is subscribed to, to check if they are the last owner of any of them.
+	Subscriptions.db.findByUserId(userId).forEach((subscription) => {
+		// DMs don't have owners
+		if (subscription.t === 'd') {
+			return;
+		}
+
+		const roomData = {
+			rid: subscription.rid,
+			t: subscription.t,
+			subscribers: null,
+		};
+
+		if (hasRole(userId, 'owner', subscription.rid)) {
+			// Fetch the number of owners
+			const numOwners = getUsersInRole('owner', subscription.rid).fetch().length;
+			// If it's only one, then this user is the only owner.
+			if (numOwners === 1) {
+				// Let's check how many subscribers the room has.
+				const options = { sort: { ts: 1 } };
```

Comment:
since you're using only the `u._id` field, please change to request only it:
```suggestion
				const options = { fields: { 'u._id': 1 }, sort: { ts: 1 } };
```